### PR TITLE
Remove archived clientset

### PR DESCRIPTION
**Pull request by caesarxuchao**
We have [client-go](https://github.com/kubernetes/client-go) now, no need to keep archived clientset in master of k8s.io/kubernetes.

Also updated the client-gen README.

Fix https://github.com/kubernetes/kubernetes/issues/35803.

cc @mikedanese 

```release-note
The main kubernetes repository stops hosting archived version of released clients. Please use [client-go](https://github.com/kubernetes/client-go).
```

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35928)

### CODE REVIEW

**Comment by mikedanese** [comment]
Thanks.

File: docs/devel/generating-clientset.md [hunk 1]
```diff
@@ -7,67 +7,54 @@ cycle of the generated clientsets.
...
-adoption of clientset. We will keep updating it as our API evolves. Eventually
-it will be replaced by a versioned clientset.
-
-- release_1_2: release_1_2 clientset is a versioned clientset, it includes
-clients for the core v1 objects, extensions/v1beta1, autoscaling/v1, and
-batch/v1 objects. We will NOT update it after we cut the 1.2 release. After the
-1.2 release, we will create release_1_3 clientset and keep it updated until we
-cut release 1.3.
-
+To get stable clientsets, please refer to the [client-go
```
Comment (nikhiljindal) [inline docs/devel/generating-clientset.md]: Probably add a note saying new code should use client-go and we are in the process of updating existing code to use client-go. Perhaps also add link to the issue?

**Comment by nikhiljindal** [comment]
LGTM with one suggestion for README

File: docs/devel/generating-clientset.md (hunk 1 above)
Comment (caesarxuchao) [inline docs/devel/generating-clientset.md]: Thanks. Will do.

**Comment by caesarxuchao** [comment]
Done. Applying the lgtm. Thanks for the suggestion.

**Comment by quinton-hoole** [comment]
@k8s-bot federation gce e2e test this

**Comment by caesarxuchao** [comment]
@k8s-bot unit test this

**Comment by caesarxuchao** [comment]
@k8s-bot federation gce e2e test this

**Comment by nikhiljindal** [comment]
federation e2e timed out.
Since this PR is just deleting old clients, I think its enough to just verify that build is not broken.
So this should be fine to merge.

**Pull request merged**